spec/puppet-lint/lexer_spec.rb

```diff
@@ -139,6 +139,50 @@
       expect(tokens[2].column).to eq(8)
     end
 
+    it 'should handle a string with newline characters' do
+      @lexer.interpolate_string("foo\n    \"/bin/foo\" >>\n    /bar/baz\"", 1, 1)
+      tokens = @lexer.tokens
+
+      expect(tokens.length).to eq(3)
+
+      expect(tokens[0].type).to eq(:STRING)
+      expect(tokens[0].value).to eq("foo\n    ")
+      expect(tokens[0].line).to eq(1)
+      expect(tokens[0].column).to eq(1)
+
+      expect(tokens[1].type).to eq(:DQPOST)
+      expect(tokens[1].value).to eq('/bin/foo')
+      expect(tokens[1].line).to eq(1)
+      expect(tokens[1].column).to eq(11)
+
+      expect(tokens[2].type).to eq(:DQPOST)
+      expect(tokens[2].value).to eq(" >>\n    /bar/baz")
+      expect(tokens[2].line).to eq(2)
+      expect(tokens[2].column).to eq(20)
```

Comment:
It's getting late here, so my brain isn't quite running at full speed, but that sequence of tokens doesn't look right (unrelated to your line number change)

Reply:
I thought it was an interesting sequence.  It's possible it is a test case for a string that is impossible in reality.

I was attempting to test the case in interpolate_string where it creates tokens because get_string_segment split on double quote.

Reply:
Maybe use a heredoc instead of a quote with new lines inside it? That's visually easier for most to comprehend and help with lexing it in your head. Same for the other string.